src/vm-memory/src/mmap.rs

```diff
@@ -269,14 +339,32 @@ impl GuestMemoryMmap {
     ///
     /// Valid memory regions are specified as a slice of (Address, Size) tuples sorted by Address.
     pub fn from_ranges(ranges: &[(GuestAddress, usize)]) -> result::Result<Self, Error> {
-        Self::from_ranges_with_files(ranges.iter().map(|r| (r.0, r.1, None)))
+        Self::from_ranges_with_files(ranges.iter().map(|r| (r.0, r.1, None)), false)
+    }
+
+    /// Creates a container, allocates anonymous memory for guest memory regions and enables dirty
+    /// page tracking.
+    ///
+    /// Valid memory regions are specified as a slice of (Address, Size) tuples sorted by Address.
+    pub fn from_ranges_with_tracking(
+        ranges: &[(GuestAddress, usize)],
+    ) -> result::Result<Self, Error> {
+        Self::from_ranges_with_files(ranges.iter().map(|r| (r.0, r.1, None)), true)
     }
```

Comment:
Is this needed ? Can't we just add `track_dirty_pages` as a parameter to `from_ranges` ?

Reply:
From_ranges function has a lot of usages inside Firecracker, so adding an additional dedicated function to handle dirty page tracking helped minimize the impact on the current code.

Reply:
Can we also add a method 'enable_dirty_tracking()' here ? Instead of having a different constructor ?